flash/tabular/classification/data/data.py

```diff
@@ -342,25 +287,58 @@ def from_df(
         """
         categorical_cols, numerical_cols = cls._sanetize_cols(categorical_cols, numerical_cols)
 
-        train_df, val_df, test_df = cls._split_dataframe(train_df, val_df, test_df, val_size, test_size)
-
-        preprocess = preprocess or cls.preprocess_cls.from_data(
-            train_df,
-            val_df,
-            test_df,
-            predict_df,
-            target_col,
-            numerical_cols,
-            categorical_cols,
-            is_regression,
+        mean, std, classes, codes, target_codes = cls.compute_state(
+            train_df, val_df, test_df, predict_df, target_col, numerical_cols, categorical_cols
         )
 
-        return cls.from_load_data_inputs(
-            train_load_data_input=train_df,
-            val_load_data_input=val_df,
-            test_load_data_input=test_df,
-            predict_load_data_input=predict_df,
+        return cls.from_data_source(
+            data_source="df",
```

Comment:
I would for a more explicit naming "data_frame"

Reply:
Yeah, `data_frame` is better, cahnged it everywhere